### PR TITLE
Update output documentation

### PR DESCRIPTION
**Pull request by paldepind**
The current documentation is pretty outdated with regards to how output works. This is improves the situation somewhat.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -414,25 +414,52 @@ _output_ from it.
...
+```ts
+const usernameInput = input({ placeholder: "Username"});
+```
+
+The type of the component constructed above is as follows ( the `...`
+refer to the fact that we have omitted a lot of the output to keep
+things simple).
+
+```ts
+Component<{}, { value: Behavior<string>, click: Stream<ClickEvent>, ... }
```
Comment (limemloh) [inline README.md]: missing closing >

File: README.md [hunk 2]
```diff
@@ -443,44 +470,129 @@ const usernameInput = input({
...
+selected output. The result is that `counterView` returns a component
+that produces two streams as its output.
+
+### An analogy with promises
+
+As mentioned above using the `output` method is a bit like adding
+event listeners in other frameworks. However, there are fundamental
+differences between the two things. If you are familiar with how
+asynchronous functions that takes callbacks differ from asynchronous
+function that returns promises then the following analogy make help
```
Comment (limemloh) [inline README.md]: >the following analogy make help
I assume you mean to write:
_the following analogy **may** help_

File: README.md [hunk 3]
```diff
@@ -414,25 +414,52 @@ _output_ from it.
...
 
-* A button will output a stream of click events. So the type of a
-  button may be `Component<Stream<Event>>`
-* A input box could output a behavior of the text inside the input.
-  It's type would be `Component<Behavior<string>>`
+* A button outputs, among other things, a stream of click events. So
+  part of its output is a stream of the type `Stream<ClickEvent>>`.
+* An input box's output includes a behavior of the text inside the
+  input. The type would be `Behavior<string>`.
 * A checkbox might output a behavior representing wether it is checked
```
Comment (limemloh) [inline README.md]: >wether
should be _whether_.

Comment (limemloh) [reply README.md]: I guess that is actually an old typo, but should still be fixed 😄

**Comment by paldepind** [comment]
Thanks @limemloh :smile: I've fixed the typos.

**Pull request merged**